src/blob/core.rs

```diff
@@ -183,31 +183,54 @@ impl Blob {
         debug!("blob read any");
         let entry = self
             .get_any_entry(key, meta)
-            .await?
+            .await
+            .unwrap()
             .ok_or_else(|| Error::from(ErrorKind::RecordNotFound))?;
         debug!("blob read any entry found");
         let buf = entry
             .load()
             .await
-            .with_context(|| format!("failed to read key {:?} with meta {:?}", key, meta))?
+            .with_context(|| format!("failed to read key {:?} with meta {:?}", key, meta))
+            .unwrap()
             .into_data();
         debug!("blob read any entry loaded bytes: {}", buf.len());
         Ok(buf)
     }
 
     #[inline]
-    pub(crate) fn read_all<'a>(&'a self, key: &'a [u8]) -> Entries<'a> {
-        self.index.get_entries(key, self.file.clone())
+    pub(crate) async fn read_all(&self, key: &[u8]) -> Result<Option<Vec<Entry>>> {
+        let headers = self.index.get_all(key).await?;
+        Ok(headers.map(|h| {
+            debug!("blob core read all {} headers", h.len());
+            Self::headers_to_entries_with_meta(h, self.file.clone())
+        }))
+    }
+
+    fn headers_to_entries_with_meta(headers: Vec<RecordHeader>, file: File) -> Vec<Entry> {
+        headers
+            .into_iter()
+            .map(|header| Entry::new(header, file.clone()))
+            .collect()
     }
 
     async fn get_any_entry(&self, key: &[u8], meta: Option<&Meta>) -> Result<Option<Entry>> {
```

Comment:
it looks like to get particular entry with matching meta but not an any. Should we rename it to `get_entry`

Reply:
Agree